### PR TITLE
[REVIEW] Fixes `row_lexicographic_comparator` issue with handling two tables

### LINKED ISSUE #3207: [BUG] `row_lexicographic_comparator` only works when left and right are the same table
**Describe the bug**
The row_lexicographic_comparator in cudf/table/row_operators.cuh swaps the left and right indices when the column order is descending, but this only works when the same table is used for left and right.
```
      if (not ascending) {
        thrust::swap(lhs_index, rhs_index);
      }
```

**Expected behavior**
The column order should instead determine how the weak ordering returned by the comparator is handled.

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
Fixes `row_lexicographic_comparator` issue with handling two tables when columns can be `DESCENDING` order.

This has test cases for both to check this scenario and other generic test cases in sort set of test cases.

close #3207

### CODE REVIEW

File: cpp/tests/sort/sort_test.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+template <typename T>
+struct SortedOrder : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(SortedOrder, cudf::test::NumericTypes);
+
+TYPED_TEST(SortedOrder, WithNullMax)
+{
+    using T = TypeParam;
+
+    cudf::test::fixed_width_column_wrapper<T> col1{{5, 4, 3, 5, 8}, {1, 1, 0, 1, 1}};
```
Comment (jrhemstad) [inline cpp/tests/sort/sort_test.cpp]: Maybe add a test that has a few equal rows.

File: cpp/tests/table/table_view_tests.cu [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+{
+    std::vector<cudf::column_view> cols{};
+
+    cudf::table_view input(cols);
+    cudf::size_type expected = 0;
+
+    EXPECT_EQ(input.num_columns(), expected);
+}
+
+TEST_F(TableViewTest, TestLexiologicalComparatorTwoTableCase)
```
Comment (harrism) [inline cpp/tests/table/table_view_tests.cu]: Lexiological? Do you mean `Lexicographical`?

Comment (rgsl888prabhu) [reply cpp/tests/table/table_view_tests.cu]: My bad

**Comment by rgsl888prabhu** [comment]
rerun tests

**Pull request merged**